mono/metadata/w32handle.c

```diff
@@ -1284,13 +1284,6 @@ mono_w32handle_wait_multiple (gpointer *handles, gsize nhandles, gboolean waital
 	if (timeout != MONO_INFINITE_WAIT)
 		start = mono_msec_ticks ();
 
-	for (i = 0; i < nhandles; ++i) {
```

Comment:
Here is the scenario I was thinking of:

a) Thread A creates some SafeHandles tied to w32handles, and shared it with thread B
b) Thread A waits on the SafeHandles, begins to execute icall code but has not inc'ed yet
c) Thread B disposes one of the shared SafeHandles or calls `_unref` directly (or, there is an accidental double `_unref`, etc.) The handle is eagerly destroyed and zeroed for reuse
d) Say Thread C is constantly thrashing handles, causing the allocation scanning to wrap around the handle heap many times. The empty handle slot may be scanned and filled with a fully valid new handle very quickly
e) Thread A finally gets to here. We are now ref'ing and validating the wrong handle